src/compat/e-129/ArkoudaRegexCompat.chpl

```diff
@@ -0,0 +1,30 @@
+module ArkoudaRegexCompat {
+  import Regex.regex as chapelRegex;
+  import Regex.regexMatch;
+  import Regex.compile;
+
+  record regex {
```

Comment:
We don't want to compile the regex for each search/match. I think since we couldn't throw in <=1.29 initializers, let's just stick with the `compile` interface for now (and add a comment here about why it's needed)